clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -4849,9 +4852,12 @@ CodeGenModule::GetOrCreateLLVMGlobal(StringRef MangledName, llvm::Type *Ty,
         Entry->setLinkage(llvm::Function::ExternalLinkage);
     }
 
-    // Handle dropped DLL attributes.
-    if (D && !D->hasAttr<DLLImportAttr>() && !D->hasAttr<DLLExportAttr>() &&
-        !shouldMapVisibilityToDLLExport(D))
+    // Handle dropped dllimport.
+    if (D &&
+        (Entry->getDLLStorageClass() ==
+         llvm::GlobalVariable::DLLImportStorageClass) &&
+        !D->hasAttr<DLLImportAttr>() &&
+        !shouldMapVisibilityToDLLExport(cast_or_null<NamedDecl>(D)))
```

Comment:
```suggestion
        !shouldMapVisibilityToDLLExport(cast<NamedDecl>(D)))
```
D is non-null here.

Reply:
This part of the code isn't directly related to the change but I assume I can go ahead and change it?

Reply:
Sorry, the above comment should have been in reply to the instance above. This cast shouldn't be needed at all. I suspect a copy & paste error on my behalf!

Reply:
I've fixed up both instances of the call to `shouldMapVisibilityToDLLExport()`.